src/pybind/mgr/requirements-mgr.txt

```diff
@@ -0,0 +1,6 @@
+CherryPy==18.8.0
+pyjwt==2.4.0
+routes==2.5.1
+werkzeug==2.0.3
+saml==0.9.0
+setuptools==59.6.0
```

Comment:
As now these packages are going to be installed and released by ourselves from PyPI packages (instead of from distro packages), it's key that we ensure a similar level of trust/security as the distro package managers (Supply Chain Management attacks are the big thing these days).

Based on [the PyPI security recomendations](https://pip.pypa.io/en/stable/topics/secure-installs/), we should always:
* Install binary distributions (instead of source code packages) via `--only-binary :all:`,
* Force hash-checking via `requirements.txt` with:
```
<package> == <x.y.z> \
    --hash:--hash=sha256:<hash_1> \
    --hash:--hash=sha256:<hash_2>
```

Reply:
What does that mean for ppc64le, s390x, etc?

Reply:
@ktdreyer do you mean the hashing? For platform-independent packages the hash should be exactly the same. For platform-specific (compiled) packages (which I'm not sure if we have such) [you could specify multiple hashes](https://pip.pypa.io/en/stable/topics/secure-installs/#multiple-hashes-per-package).

Reply:
Thanks. For example packages like `cryptography` have Rust-based parts now, so they have per-arch binaries. I'm wondering if we want to build these from source.

Reply:
I don't think it's needed. Many Python projects with compiled sources or compiled deps (including `cryptography`) haved moved to the [`manylinux` wheels](https://peps.python.org/pep-0513/).

As we're not cross-compiling, a [`pip install/download --only-binary :all:`](https://pip.pypa.io/en/stable/cli/pip_install/#cmdoption-only-binary) should provide what we need for each platform.

For example, I run that in the following distros `centos:stream8` and `ubuntu:20.04` and the binary packages installed are the same: 
* `cryptography-39.0.0-cp36-abi3-manylinux_2_17_x86_64.manylinux2014_x86_64.whl`
* `cffi-1.15.1-cp36-cp36m-manylinux_2_5_x86_64.manylinux1_x86_64.whl`
* `pycparser-2.21-py2.py3-none-any.whl`

Only thing needed is to update `pip` to the same version (since packaged pip versions usually differ).

Reply:
Ok. I think the difference here is that with RPM packages, Red Hat maintains the ability to easily patch and rebuild the dependencies in the event of security vulnerability that upstream cannot or will not fix. When we depend on pre-built binaries, that's no longer the case, and we're relying on the goodwill of the upstream authors to do that release engineering work for us.

Reply:
On paper it makes a lot of sense, but the reality is that for some/many of these packages we (and when I say "we" I mean "you" :P) are the maintainers... I'm thinking, for example, of `cherrypy/cheroot`. And that applies to many of the non-core Python packages.

Additionally, there's no long-term maintenance/bug-fixing commitment for extended repos (EPEL in the CentOS world, or Universe for Debian/Ubuntu).

With Github Dependabot alerts, we receive immediate notifications and PRs fixing detected vulnerable packages. And we may also ensure that on Ceph upgrade, all our Python deps would be updated to the same (hopefully safest) version.

Is there any blindspot around that I'm missing? 🤔